samples/minimum_criteria.rst

```diff
@@ -0,0 +1,20 @@
+.. _definition_and_minimum_criteria:
+
+Definition & Minimum Criteria
+#############################
+
+A Sample is an application in Zephyr that briefly showcases how the functionality of domain/subsystem/feature is used.
```

Comment:
@laurenmurphyx64  Added some of the suggested changes. Please re-review again. For the definition of the samples, I still think it is a good idea to use "briefly showcases" terminology to indicate a clear distinction between a sample and a demo(which would be a more elaborate illustration). Let me know if it doesn't make sense. Also, for leveraged in CI part, the intention was to communicate that even though samples are being tested in CI, we shall not confuse them with testcases because its purpose is to show how a feature is used but we are still testing samples in CI for quality confidence.  Hope that makes sense. Let me know your thoughts.